src/System.Management.Automation/engine/SessionState.cs

```diff
@@ -65,6 +66,9 @@ internal SessionStateInternal(SessionStateInternal parent, bool linkToGlobal, Ex
             // is used for the pushd and popd commands
 
             _workingLocationStack = new Dictionary<String, Stack<PathInfo>>(StringComparer.OrdinalIgnoreCase);
+            // conservative choice to limit the memory impact in case of a regression
```

Comment:
I would put a space before the comment. I'd also add a comment saying that this is for the Set-Location history to distinguish it from the earlier "pushd and popd" comment.